test/tests/performance/pbs_node_buckets_perf.py

```diff
@@ -66,69 +69,97 @@ def cust_attr_func(self, name, totalnodes, numnode, attribs):
         self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
-                            {'scheduling': 'False'})
+                            {'scheduling': 'False'}, expect=True)
 
-        # wait for cycle to finish
-        self.scheduler.log_match("Leaving Scheduling Cycle", starttime=t,
-                                 max_attempts=120, interval=5)
+        # 600 * 2sec = 20m which is the max cycle length
+        self.server.expect(SERVER, {'server_state': 'Scheduling'}, op=NE,
+                           max_attempts=600, interval=2)
         c = self.scheduler.cycles(lastN=1)[0]
-        cycle1_time = c.end - c.start
-
-        a = {'Resource_List.place': 'scatter:excl'}
-        for jid in jids:
-            self.server.alterjob(jid, a)
+        return c.end - c.start
 
-        t = int(time.time())
+    def compare_normal_path_to_buckets(self, place, num_jobs):
+        """
+        Submit num_jobs jobs and run two cycles.  First one with the normal
+        node search code path and the second with buckets.  Print the
+        time difference between the two cycles.
+        """
+        # Submit one job to eat up the resources.  We want to compare the
+        # time it takes for the scheduler to attempt and fail to run the jobs
+        a = {'Resource_List.select': '1429:ncpus=1:color=yellow',
+             'Resource_List.place': place,
+             'Resource_List.walltime': '1:00:00'}
+        Jyellow = Job(TEST_USER, attrs=a)
+        Jyellow.set_sleep_time(3600)
+        jid_yellow = self.server.submit(Jyellow)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid_yellow)
 
-        # run only one cycle
-        self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
-                            {'scheduling': 'True'})
         self.server.manager(MGR_CMD_SET, MGR_OBJ_SERVER,
-                            {'scheduling': 'False'})
+                            {'scheduling': 'False'}, expect=True)
 
-        # wait for cycle to finish
-        self.scheduler.log_match("Leaving Scheduling Cycle", starttime=t,
-                                 max_attempts=120, interval=5)
+        # Shared jobs use standard code path
+        a = {'Resource_List.select':
+             '1429:ncpus=1:color=blue+1429:ncpus=1:color=yellow',
+             "Resource_List.place": place}
+        jids = self.submit_jobs(a, num_jobs)
+
+        cycle1_time = self.run_cycle()
+
+        # Excl jobs use bucket codepath
+        a = {'Resource_List.place': place + ':excl'}
+        for jid in jids:
+            self.server.alterjob(jid, a)
+
+        cycle2_time = self.run_cycle()
 
-        c = self.scheduler.cycles(lastN=1)[0]
-        cycle2_time = c.end - c.start
         self.logger.info('Cycle 1: %d Cycle 2: %d Cycle time difference: %d' %
                          (cycle1_time, cycle2_time, cycle1_time - cycle2_time))
-        self.assertTrue(cycle1_time > cycle2_time)
+        self.assertGreater(cycle1_time, cycle2_time)
```

Comment:
Hi @bhroam , Can you make use of set_test_measurements API so that the test results are stored in the JSON output file .
self.set_test_measurements({"test_measure": cycle_time_without_node_buckets,"value": cycle_time})
self.set_test_measurements({"test_measure": cycle_time_with_node_buckets,"value": cycle2_time}).
We are planning to store all results from all the performance tests to json file , So we can use them to compare performance across releases.